llvm/include/llvm/CodeGen/TargetInstrInfo.h

```diff
@@ -113,9 +113,17 @@ struct ExtAddrMode {
 ///
 class LLVM_ABI TargetInstrInfo : public MCInstrInfo {
 protected:
+  /// Subtarget specific sub-array of MCInstrInfo's RegClassByHwModeTables
+  /// (i.e. the table for the active HwMode). This should be indexed by
+  /// MCOperandInfo's RegClass field for LookupRegClassByHwMode operands.
+  const int16_t *const RegClassByHwMode;
+
+public:
```

Comment:
Why did this constructor become public?

Reply:
Because I create a lot of merge conflicts for myself